test/acceptance/tests/basic/terraform/basic-install/main.tf

```diff
@@ -281,10 +281,6 @@ module "test_server" {
   acl_secret_name_prefix         = var.suffix
   consul_ecs_image               = var.consul_ecs_image
 
-  // Test passing both a role resource and role data source objects to make sure both
-  // have the necessary fields ("arn" and "id").
-  task_role                     = aws_iam_role.task
-  execution_role                = aws_iam_role.execution
```

Comment:
Why remove these?

Reply:
They were causing the `secure` case to fail on the `terraform apply`. I'm trying to figure out exactly why. I wanted to get the rest of the PR out for review while I worked on figuring this out.

Reply:
Oh, right. Because the execution role needs permission to fetch AWS Secrets Manager secrets in the secure case, and those perms aren't set by basic-install.

I think it's fine to leave this for a follow up, as long as we don't lose track of it.

Reply:
Well, wait, but mesh-task should create a policy for those additional permissions:

https://github.com/hashicorp/terraform-aws-consul-ecs/blob/b8444236a962711e6070bd420723b2d07b99d8e6/modules/mesh-task/iam.tf#L71 

So actually I'm still not sure why it would fail.

Reply:
It fails due to a difference in the way that splat expressions are handled between Terraform v1.0 and v1.1. The CI pipeline uses v1.0 and I was using v1.1 locally. We will leave the test the as proposed in the PR for now and fix this issue in a future PR.